src/votekit/ballot_generator.py

```diff
@@ -39,32 +41,89 @@ def __init__(
         self.ballot_length = (
             ballot_length if ballot_length is not None else len(candidates)
         )
-        if isinstance(candidates, dict):
-            self.slate_to_candidate = candidates
-            self.candidates = list(
-                {cand for cands in candidates.values() for cand in cands}
+        self.candidates = candidates
+
+        if bloc_voter_prop and pref_interval_by_bloc:
+            if round(sum(bloc_voter_prop.values())) != 1:
+                raise ValueError("Voter proportion for blocs must sum to 1")
+            for interval in pref_interval_by_bloc.values():
+                if round(sum(interval.values())) != 1:
+                    raise ValueError("Preference interval for candidates must sum to 1")
+            if bloc_voter_prop.keys() != pref_interval_by_bloc.keys():
+                raise ValueError("slates and blocs are not the same")
+
+            self.pref_interval_by_bloc = pref_interval_by_bloc
+            self.bloc_voter_prop = bloc_voter_prop
+
+    @classmethod
+    def from_params(
```

Comment:
I like this, its more user friendly then passing in a dictionary full of stuff. If a user sets the intervals when initializing are those overwritten when this method is called?

Reply:
yes